server/storage/mvcc/watchable_store.go

```diff
@@ -482,6 +484,26 @@ func (s *watchableStore) progress(w *watcher) {
 	}
 }
 
+func (s *watchableStore) progressAll(watchers map[WatchID]*watcher) bool {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	// Any watcher unsynced?
+	for w, _ := range watchers {
+		if _, ok := s.synced.watchers[watchers[w]]; !ok {
+			return false
+		}
+	}
+
+	// If all watchers are synchronised, send out progress notification
+	// on first watcher
```

Comment:
nit and optional: describing that the broadcasting mechanism is implemented in [client side](https://github.com/etcd-io/etcd/blob/main/client/v3/watch.go#L762) might be a little bit friendly? No strong preference at all.